### PR TITLE
[E2E test coverage]: Disable block product editor

### LINKED ISSUE #39417: [E2E test coverage]: Disable block product editor
Add E2E tests to disable the new product form.

This will test if it:
- _can disable the new product form_

### PR DESCRIPTION
**Pull request by octaedro**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

This PR introduces E2E tests to ensure that disabling the new experience works as expected.

Closes #39417.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Execute the standard commands to launch the local E2E environment.
2. Make sure the test is not flaky by running multiple instances of it repeatedly. You can do so through this command:
    ```bash
    cd plugins/woocommerce
    pnpm test:e2e-pw disable-block-product-editor.spec.js --retries=0 --workers=4 --repeat-each=8
    ```
4. All instances of this test should pass, which indicates it isn't flaky.

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/tests/e2e-pw/tests/merchant/products/block-editor/disable-block-product-editor.spec.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+	clickAddNewMenuItem,
+	expectBlockProductEditor,
+	expectOldProductEditor,
+	isBlockProductEditorEnabled,
+	toggleBlockProductEditor,
+} = require( '../../../../utils/simple-products' );
+
+const ALL_PRODUCTS_URL = 'wp-admin/edit.php?post_type=product';
+const NEW_EDITOR_ADD_PRODUCT_URL =
+	'wp-admin//admin.php?page=wc-admin&path=%2Fadd-product';
```
Comment (lanej0) [inline plugins/woocommerce/tests/e2e-pw/tests/merchant/products/block-editor/disable-block-product-editor.spec.js]: Probably not going to cause an issue, but you have an extra `/` in the URL here.

File: plugins/woocommerce/tests/e2e-pw/utils/simple-products.js [hunk 2]
```diff
@@ -54,7 +56,31 @@ async function clickAddNewMenuItem( page ) {
 		.click();
 }
 
+/**
+ * This function checks if the old product editor is visible.
+ *
+ * @param {Page} page - Represents a single tab in a browser, controlled by puppeteer.
```
Comment (lanej0) [inline plugins/woocommerce/tests/e2e-pw/utils/simple-products.js]: {nb} But the comment references puppeteer... and this is Playwright :-)

**Comment by octaedro** [comment]
Thank you @lanej0 for the review.
I just addressed the changes you suggested, could you take another look at this PR?

**Comment by nigeljamesstevenson** [comment]
Hi @octaedro ! Jon was out AFK but I notice there are conflicts in the branch - if you want a quick check on this feel free to resolve and ping me - thanks a lot!

**Review by lanej0 — APPROVED** [review]
Looks good!

**Pull request merged**